src/dashboard/store/modules/metamask.js

```diff
@@ -0,0 +1,76 @@
+let networks = {
+  '-1': { color: 'warning' },
+  1: { name: 'Main Ethereum Network', color: 'green' },
+  3: { name: 'Ropsten Test Network', color: 'red' },
+  4: { name: 'Rinkeby Test Network', color: 'orange' },
+  42: { name: 'Kovan Test Network', color: 'purple' },
+};
+let donate;
+
+export default {
+  namespaced: true,
+
+  state: {
+    netId: -1,
+    address: null,
+  },
+  getters: {
+    netName(state) {
+      return networks[state.netId].name;
+    },
+    netColor(state) {
+      return networks[state.netId].color;
+    },
+    isAddress() {
+      return addr => donate.isAddress(addr);
+    },
+  },
+  actions: {
+    async initialize({ dispatch }) {
+      let module = await import('../../../lib/donate.js');
+      donate = new module.default();
+      setTimeout(() => dispatch('update'), 500);
+      setInterval(() => dispatch('update'), 5000);
+    },
+    async update({ commit }) {
+      try {
+        let id = await donate.getId();
+        commit('setNetId', id);
+        let addr = await donate.getMyAddress();
+        if (addr !== undefined) {
+          commit('setAddress', addr);
+        } else {
+          commit('unsetAddress');
+        }
+      } catch (err) {
+        console.error('Failed to update metamask status:', err);
+        commit('unsetNetId');
+        commit('unsetAddress');
+      }
+    },
+    async donateAll({ dispatch }, donations) {
+      let all = await donate.donateAll(donations);
+      all.map(async d => {
+        let donation = await d;
```

Comment:
Don't we want to do this in "parallel" with Promise.all? Or maybe that's overkill until we go above 12 donations.

Reply:
Nope, we absolutely don't want `Promise.all` since it fails early.